lib/hdf.py

```diff
@@ -35,3 +36,95 @@ def get_returnn_simple_hdf_writer(returnn_root: Optional[str]):
     from returnn.datasets.hdf import SimpleHDFWriter
 
     return SimpleHDFWriter
+
+
+class NextGenHDFWriter:
+    """
+    This class is a helper for writing the of returnn NextGenHDFDataset
+    """
+
+    def __init__(
+        self,
+        filename: str,
+        label_info_dict: Dict,
+        feature_names: Optional[List[str]] = None,
+        label_data_type: type = np.uint16,
+        label_parser_name: str = "sparse",
+        feature_parser_name: str = "feature_sequence",
+    ):
+        """
+        :param label_info_dict: a dictionay with the label targets used in returnn training as key and numebr of label classes as value
+        :param feature_names: additional feature data names
+        :param label_data_type: type that is used to store the data
+        :param label_parser_name: this should be checked against returnn implementations
+        "param feature_parser_name: as above
+        """
+        self.label_info_dict = label_info_dict
+        self.label_parser_name = label_parser_name
+        self.feature_names = feature_names
+        if feature_names is not None:
+            self.feature_parser_name = feature_parser_name
+        self.label_data_type = label_data_type
+        self.string_data_type = h5py.special_dtype(vlen=str)
+        self.sequence_names = []
+        self.group_holder_dict = {}
+
+        self.file_init()
+
+    def file_init(self):
+        self.temp_file = tempfile.NamedTemporaryFile(suffix="_NextGenHDFWriter_outHDF")
+        self.temp_path = self.temp_file.name
+        self.out_hdf = h5py.File(self.temp_path, "w")
+
+        logging.info(f"processing temporary file { self.temp_path}")
+
+        # root
+        self.root_group = self.out_hdf.create_group("streams")
+
+        for label_name, label_dim in self.label_info_dict.items():
```

Comment:
Following the init restructuring from above, I think it would be better to just loop here over all streams and have a separate `create_group` function that you can override as user to add your own custom streams.

Reply:
Sill, I do not know how to cleanly deal with the "label_dim" argument in that case...